lib/spack/spack/bootstrap/core.py

```diff
@@ -569,11 +640,20 @@ def ensure_winsdk_external_or_raise() -> None:
     spack.detection.update_configuration(externals, buildable=False)
 
 
+def ensure_win_resources() -> None:
+    ensure_resource("file")
+    ensure_resource("gpg")
+
+
 def ensure_core_dependencies() -> None:
     """Ensure the presence of all the core dependencies."""
     if sys.platform.lower() == "linux":
         ensure_patchelf_in_path_or_raise()
-    if not IS_WINDOWS:
+    if IS_WINDOWS:
+        # Windows also requires gpg (as in the check below)
+        # but that is resolved by this call
+        ensure_win_resources()
```

Comment:
I was hoping to also fetch these resources with the `spack bootstrap now` command which invokes this `ensure_core_dependencies` method.
So necessary? No. That said I think we should leave it. We won't ever be "rebootstrapping" any of these so I don't think it hurts to have here?

Reply:
> I don't think it hurts to have here?

I don't think it will cause problems directly, but my concern is more that if we set up multiple pathways for something to be addressed, it will not be clear which one is the required one. Someone could come along later, notice both of these, then remove the other. In general, if there is a 100% solution and a less-than-100% pathway that also exists, I prefer to remove the latter.

Reply:
To be clear, I am not opposed to removing this, but I would like to understand why this is a different case from something like GPG  or Clingo, where we do have multiple paths to bootstrap them so I can make the right call in the future. For those two, we have the `bootstrap now` approach, which invokes the method currently in question, and then the "normal spack flow" approach, where a spack command encounters the need for clingo/gpg and Spack calls into the bootstrap at that point. They end up calling the same methods under the hood, but so does this.

Reply:
I didn't see there were multiple pathways for Clingo. IMO Clingo bootstrapping is distinct in that it could take a significant amount of time: one of the automated fallback bootstrapping mechanisms is to install from source; this is true to a lesser extent for GPG on Unix as well. On Windows, these bootstrapping operations are always fast (because downloading the binaries is the only option).

For that reason, it's ideal to bootstrap Clingo/GPG up front (in violation of the general principle of only having one pathway for things).